client_context.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from decimal import Decimal
+from typing import Optional
+import json
+
+from edge_service import EdgeService
+from featurehub_repository import FeatureHubRepository
+from fh_state_base_holder import FeatureStateBaseHolder
+from strategy_attribute_country_name import StrategyAttributeCountryName
+from strategy_attribute_device_name import StrategyAttributeDeviceName
+from strategy_attribute_platform_name import StrategyAttributePlatformName
+
+
+class ClientContext:
+    """holds client context"""
+
+    def __init__(self, repo: FeatureHubRepository, edge: EdgeService):
```

Comment:
this base class doesn't require the edge service, only the subclasses do

Reply:
ok, i will tidy this up in the next PR